Towny/src/main/java/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -2440,32 +2440,50 @@ public static void townBuy(CommandSender sender, String[] split, @Nullable Town
 		if (!TownyEconomyHandler.isActive())
 			throw new TownyException(Translatable.of("msg_err_no_economy"));
 
+		if (split.length == 0 || split[0].equalsIgnoreCase("?") || split[0].equalsIgnoreCase("help")) {
+			HelpMenu.TOWN_BUY.send(sender);
+			return;
+		}
+
 		if (town == null && sender instanceof Player player) {
-			checkPermOrThrow(player, PermissionNodes.TOWNY_COMMAND_TOWN_BUY.getNode());
+			checkPermOrThrow(player, PermissionNodes.TOWNY_COMMAND_TOWN_BUY.getNode(split[0].toLowerCase(Locale.ROOT)));

```

Comment:
```suggestion
			checkPermOrThrow(player, PermissionNodes.TOWNY_COMMAND_TOWN_BUY.getNode());

```

Probably not good to test the subcommand node, any addon plugin might not set that up.